pkg/scheduler/framework/runtime/framework.go

```diff
@@ -246,7 +238,7 @@ func WithClusterEventMap(m map[framework.ClusterEvent]sets.String) Option {
 var _ framework.Framework = &frameworkImpl{}
 
 // NewFramework initializes plugins given the configuration and the registry.
-func NewFramework(r Registry, plugins *config.Plugins, args []config.PluginConfig, opts ...Option) (framework.Framework, error) {
+func NewFramework(r Registry, pr config.KubeSchedulerProfile, opts ...Option) (framework.Framework, error) {
```

Comment:
did you consider changing pr to be a pointer? Also, the variable should be named profile. It's a big function.